pipeline_steps/common.groovy

```diff
@@ -968,7 +968,12 @@ void use_node(String label=null, body){
       print "${env.NODE_NAME} preparation complete, now ready for use."
       body()
     } catch (e){
-      print "Caught exception on ${env.NODE_NAME}: ${e}"
+      errString = "Caught exception on ${env.NODE_NAME}: ${e} Build: ${env.BUILD_URL}"
+      print errString
+
+      if (env.NODE_NAME =~ /^nodepool-/ && env.HOLD_ON_ERROR.asBoolean() == true){
```

Comment:
Given that `HOLD_ON_ERROR` will be a string, I think you want to use [toBoolean](http://docs.groovy-lang.org/latest/html/api/org/codehaus/groovy/runtime/StringGroovyMethods.html#toBoolean-java.lang.String-) instead of [asBoolean](http://docs.groovy-lang.org/latest/html/api/org/codehaus/groovy/runtime/StringGroovyMethods.html#asBoolean-java.lang.CharSequence-).

I wonder if we could be doing a better job with this type of data. Some variables in env are only required by pipeline_steps. Maybe we should defined global variables for these things at the beginning and delete them from `env` so they can be appropriately typed and used without requiring special understanding.

Reply:
Good catch, removed the boolean var so its no longer an issue. I agree that typing is suboptimal, especially as JJB has a bool type, but stores the result as a string (As env vars are untyped). Worth revisiting this as part of a transition away from jjb.